### PR TITLE
[UnitTest] Testing the startTransactionExit function

### PR DESCRIPTION
**Pull request by fleupold**
Test cases will be:
1) Working start of transaction exit
2) Cannot withdraw unowned UTXO
3) Cannot use someone else signature
4) Cannot withdraw with bad merkleproof

### CODE REVIEW

**Comment by fleupold** [comment]
This still needs a bit more work before it's fully ready for review.

File: test/utilities.js [hunk 1]
```diff
@@ -50,6 +58,64 @@ const BlockType = {
...
+const generateMerkleTree = function(...args) {
+  const txs = Array(2**16).fill(sha3(0x0))
+  for (let i=0; i<args.length; i+=2) {
+    txs[args[i]] = args[i+1]
+  }
+  return new MerkleTree(txs, sha3)
+}
+
+const rlpEncodeTransaction = function(exitor, token, amount, inputCount, oindex) {
+  const list = [inputCount, null, null, null, null, null, token, null, null, null, null]
```
Comment (bh2smith) [inline test/utilities.js:108]: What's the purpose of all the null's here? Is this just they way it is in some smart Contract somewhere?

File: test/utilities.js [hunk 2]
```diff
@@ -50,6 +58,64 @@ const BlockType = {
...
+ * this function generates and returns the concatenated double signature.
+ */
+const generateDoubleSignature = async function(tx, tree, signer) {
+  const confirmationHash = sha3(toHex(Buffer.concat([tx.txHash, tree.getRoot()])))
+  const confSignature = await web3.eth.sign(signer, toHex(confirmationHash))
+  return tx.txSignature + confSignature.slice(2)
+}
+
+/**
+ * Given a sequence of index1, elements1, ..., indexN elementN this function returns
```
Comment (bh2smith) [inline test/utilities.js:96]: sweet idea!

**Review by bh2smith — APPROVED** [review]
This looks really good! I especially enjoy the helper functions that were included here in utilities! Great stuff.

File: test/utilities.js (hunk 1 above)
Comment (fleupold) [reply test/utilities.js:108]: As far as I could tell the indices were not used so far in the RLP representation. We might have to adjust that later, but for now we only care about index 0, 6, 7, 8, 9.

@josojo might have more background on the RLP encoding.

**Pull request merged**